### PR TITLE
fix(footer): rhdc bugs

### PR DESCRIPTION
**Pull request by heyMP**
## What I did

- #488 
  - [x] Branch test
  - [x] Code review
- #496
  - [x] Branch test
  - [x] Code review
- #498
  - [x] Branch test
  - [x] Code review
- #499
  - [x] Branch test
  - [x] Code review
- #524
  - [x] Branch test
  - [x] Code review
- #543
  - [x] Branch test
  - [x] Code review
- #534 
  - [x] Branch test
  - [x] Code review
- #552
  - [x] Branch test
  - [x] Code review
- #563 
  - [x] Branch test
  - [x] Code review

## Testing Instructions

See individual issues above for testing instructions.

### CODE REVIEW

File: elements/rh-footer/demo/rh-footer.html [hunk 1]
```diff
@@ -22,122 +22,86 @@
...
   }
+
+  *, *:before, *:after {
+    box-sizing: border-box;
+  }
 </style>
 
 <section id="kitchen-sink">
   <h2>Kitchen Sink:</h2>
-  <rh-demo>
```
Comment (heyMP) [inline elements/rh-footer/demo/rh-footer.html:29]: The only change in this file is removing <rh-demo>

**Comment by paigenotfound404** [comment]
Engage page example on Eloqua: https://engage.redhat.com/discover-more

Engage page example on Drupal, in FTS as "universal navigation" https://www.redhat.com/zh/engage/9reasons-to-explore-rhel9-20220922 and https://www.redhat.com/fr/engage/site-reliability-engineer-20220923

File: elements/rh-footer/RhFooter.ts [hunk 2]
```diff
@@ -93,7 +94,7 @@ export class RhFooter extends LitElement {
   }
 
   override render() {
-    const isMobile = this.#matchMedia.mediaQueryList?.matches ?? false;
+    const isMobile = !this.#matchMedia.mediaQueryList?.matches ?? false;
```
[the remark this answers is not in the document]
Comment (heyMP) [reply elements/rh-footer/RhFooter.ts]: Nice catch!  You're right, `!` ensures that it's boolean.

**Review by brianferry — APPROVED** [review]
Looks good to me, great job @heyMP ! Footer is looking great

**Pull request merged**